### DISCUSSION TITLE
Feature Request: disable overflow on categorical axes

### DISCUSSION THREAD

**Discussion by bendavid** [scikit-hep/boost-histogram#694] [Ideas]
Would be useful to be able to disable overflow on IntCategory and StrCategory axes
(especially when setting bin contents directly such that the overflow is not relevant by construction)

**Comment by henryiii** [comment]
This might be mostly alleviated by https://github.com/scikit-hep/boost-histogram/pull/689 - at least the bin contents part?

**Reply by bendavid** [reply]
Mmm, but isn't this the desired behaviour?  Ie when accessing a category index with a value not in the category list, one gets the overflow bin.

What I was asking for is to have the possibility to pass overflow=False when constructing an IntCategory or StrCategory axis.

My main motivation is that I want to save the extra memory.  This can matter a lot if you have e.g. a large 3d histogram where the last axis is a categorical axis with 2 or 3 categories.

**Reply by bendavid** [reply]
(the c++ library should already support this just fine)

**Reply by henryiii** [reply]
No, you can ask for the overflow bin specifically. It's extremely error prone to ask for a bin and get the overflow bin instead for a discrete, non-contiguous binning.

There's really nothing stopping adding this, by the way, it just adds to the binary size and compile complexity; this is already a lengthy and memory eating portion of the build. But otherwise there's nothing stopping this to the best of my knowledge, it just hasn't been added, probably largely because no one asked (before).

We should at least try adding it and seeing if it cause any noticeable increases or slowdowns, or causes a build to run out of memory again.

**Reply by henryiii** [reply]
(these have become very nice and heavily used axes types, so I'm not against at least trying it)
